pkg/condition/types/converters.go

```diff
@@ -0,0 +1,120 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+	"time"
+)
+
+func primitiveTypeConverterFunc[T any](value any) (any, error) {
+	v, ok := value.(T)
+	if !ok {
+		return nil, fmt.Errorf("unexpected generic type value '%q', expected '%T'", reflect.TypeOf(value), *new(T))
+	}
+
+	return v, nil
+}
+
+func numericTypeConverterFunc[T int64 | uint64 | float64](value any) (any, error) {
+	v, ok := value.(T)
+	if ok {
+		return v, nil
+	}
+
+	floatValue, ok := value.(float64)
+	bigFloat := big.NewFloat(floatValue)
+	if !ok {
+		stringValue, ok := value.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected generic type value '%q', expected '%T'", reflect.TypeOf(value), *new(T))
```

Comment:
The usage here isn't necessarily bound to generic types of `list<type>`, `map<type>`, etc..

We may avoid using the term 'generic' for this reason. I may suggest

> unexpected type value '%q', expected '%T'

instead